builder/qemu/step_run.go

```diff
@@ -16,41 +16,45 @@ import (
 // stepRun runs the virtual machine
 type stepRun struct {
 	DiskImage bool
-}
 
-type qemuArgsTemplateData struct {
-	HTTPIP      string
-	HTTPPort    int
-	HTTPDir     string
-	OutputDir   string
-	Name        string
-	SSHHostPort int
+	atLeastVersion2 bool
+	ui              packer.Ui
 }
 
 func (s *stepRun) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	config := state.Get("config").(*Config)
 	driver := state.Get("driver").(Driver)
-	ui := state.Get("ui").(packer.Ui)
+	s.ui = state.Get("ui").(packer.Ui)
 
-	// Run command is different depending whether we're booting from an
-	// installation CD or a pre-baked image
-	bootDrive := "once=d"
```

Comment:
This logic didn't need to live here; it can just live where the boot-drive arg is added in the defaults.